build.sbt

```diff
@@ -102,45 +111,48 @@ lazy val sysops = crossProject(JVMPlatform).withoutSuffixFor(JVMPlatform)
     },
   )
 
-lazy val config = crossProject(JVMPlatform).withoutSuffixFor(JVMPlatform)
-  .in(file("scalafmt-config")).settings(
+lazy val config = crossProject(JVMPlatform, NativePlatform)
+  .withoutSuffixFor(JVMPlatform).in(file("scalafmt-config")).settings(
     moduleName := "scalafmt-config",
     description := "Scalafmt config parsing",
     scalacOptions ++= scalacJvmOptions.value,
     libraryDependencies ++= Seq(metaconfig.value),
   ).jvmSettings(libraryDependencies ++= Seq(metaconfigTypesafe.value))
+  .nativeSettings(libraryDependencies ++= Seq(metaconfigSconfig.value))
 // .jsSettings(
 //   libraryDependencies ++= Seq(
 //     metaconfigHocon.value,
 //   )
 // )
 
-lazy val core = crossProject(JVMPlatform).in(file("scalafmt-core")).settings(
-  moduleName := "scalafmt-core",
-  buildInfoSettings,
-  scalacOptions ++= scalacJvmOptions.value,
-  libraryDependencies ++= Seq("org.scalameta" %% "mdoc-parser" % mdocV),
-  libraryDependencies ++= {
-    CrossVersion.partialVersion(scalaVersion.value) match {
-      case Some((2, 13)) => Seq()
-      case _ => Seq(compilerPlugin(
-          "org.scalamacros" % "paradise" % "2.1.1" cross CrossVersion.full,
-        ))
-    }
-  },
-)
+lazy val core = crossProject(JVMPlatform, NativePlatform)
+  .in(file("scalafmt-core")).settings(
+    moduleName := "scalafmt-core",
+    buildInfoSettings,
+    scalacOptions ++= scalacJvmOptions.value,
+    libraryDependencies ++= Seq("org.scalameta" %%% "mdoc-parser" % mdocV),
+    libraryDependencies ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, 13)) => Seq()
+        case _ => Seq(compilerPlugin(
+            "org.scalamacros" % "paradise" % "2.1.1" cross CrossVersion.full,
+          ))
+      }
+    },
+  )
   // .jsSettings(
   //   libraryDependencies ++= List(
   //     scalatest.value % Test // must be here for coreJS/test to run anything
   //   )
   // )
+  .nativeSettings(libraryDependencies += "com.lihaoyi" %%% "fastparse" % "3.1.0")
```

Comment:
This is probably the weirdest part. scalameta-trees seems to use fastparse in its implementation, however I think it does not point to it in the jar metadata? (I cannot find any mention of fastparse there on scaladex either). Either way, without this we get errors about missing symbols when linking.

Reply:
The annoying part here is keeping the version set to the same one scalameta uses